nixos/modules/services/misc/owncast.nix

```diff
@@ -0,0 +1,100 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let cfg = config.services.owncast;
+in {
+
+  options.services.owncast = {
+
+    enable = mkEnableOption "Whether to enable owncast";
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/owncast";
+      description = "The directory where owncast stores its data files.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for owncast.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "owncast";
+      description = "User account under which owncast runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "owncast";
+      description = "Group under which owncast runs.";
+    };
+
+    listen = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      example = "127.0.0.1";
+      description = "The IP address to bind owncast to.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 80;
+      description = ''
+        TCP port where owncast web-gui listens.
+      '';
+    };
+
+    rtmp-port = mkOption {
+      type = types.port;
+      default = 1935;
+      description = ''
+        TCP port where owncast rtmp service listens.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.owncast = {
+      wantedBy = [ "network.target" ];
+      preStart = ''
```

Comment:
😭

Is there **any** way around this?

Reply:
First of all:
Thank you for the suggestions!
I will take myself some time for them in the evening.

Unfortunally, I haven't found a way arround the preStart script by myself. I'm open to suggestions!

The application creates a static & webroot folder, BUT: some essential files are missing when doing so -> the application crashes.
When copying those folders from the source folder, everything works great.

Reply:
@aanderse With the help of @pinpox, I was able to replace the `preStart` block through `systemd.tmpfiles.rules`.